GitCommands/RevisionReader.cs

```diff
@@ -240,19 +240,30 @@ private IReadOnlyCollection<GitRevision> GetRevisionsFromArguments(GitArgumentBu
             using IProcess process = _module.GitCommandRunner.RunDetached(cancellationToken, arguments, redirectOutput: true, outputEncoding: GitModule.LosslessEncoding);
             cancellationToken.ThrowIfCancellationRequested();
 
+            // Initial buffer to give very quick feedback to user
+            List<GitRevision> revisions = new(100);
             foreach (ReadOnlyMemory<byte> chunk in process.StandardOutput.BaseStream.SplitLogOutput())
             {
                 cancellationToken.ThrowIfCancellationRequested();
 
-                if (TryParseRevision(chunk.Span, out GitRevision? revision))
+                if (TryParseRevision(chunk, out GitRevision? revision))
                 {
 #if TRACE_REVISIONREADER
                     revisionCount++;
 #endif
-                    subject.OnNext(revision);
+                    revisions.Add(revision);
+                    if (revisions.Count == revisions.Capacity)
+                    {
+                        subject.OnNext(revisions);
+
+                        // ... then use big buffer to load all the remaining revisions with better performance
```

Comment:
I am surprised this gives so much improvements.
I tried something similar before optimizing the handling, but it could be different now.
Good finding regardless!

Maybe the reading from stream can be improved too.
Right now the data is reading revision by revision. A bigger buffer could improve (there some improvements in .net8 too).
None of that made any difference when I tested though. (and --log-size  was slower than searching for '0').